lib/galaxy/schema/workflows.py

```diff
@@ -223,3 +385,642 @@ class StoredWorkflowDetailed(StoredWorkflowSummary):
+        description="The action of the tool step.",
+    )
+    license: Optional[str] = Field(
+        None,
+        title="License",
+        description="The license of the tool step.",
+    )
+    creator: Optional[str] = Field(
+        None,
+        title="Creator",
+        description="The creator of the tool step.",
+    )
+    method: str = Field(
+        ...,
+        title="Method",
+        description="The method of the tool step.",
+    )
+    enctype: str = Field(
+        ...,
+        title="Enctype",
+        description="The enctype of the tool step.",
+    )
+    tool_shed_repository: Optional[ToolShedRepositorySummary] = Field(
+        None,
+        title="Tool Shed Repository",
+        description="Information about the tool shed repository associated with the tool.",
+    )
+    link: Optional[str] = Field(
+        None,
+        title="Link",
+        description="The link of the tool step.",
+    )
+    # TODO - see lib/galaxy/tools/__init__.py - class Tool - to_dict for further typing
+    min_width: Optional[Any] = Field(
+        None,
+        title="Min Width",
+        description="The minimum width of the tool step.",
+    )
+    # TODO - see lib/galaxy/tools/__init__.py - class Tool - to_dict for further typing
+    target: Optional[Any] = Field(
+        None,
+        title="Target",
+        description="The target of the tool step.",
+    )
+
+
+class WorkflowDictPreviewStep(WorkflowDictStepsExtendedBase):
+    order_index: int = Field(
+        ...,
+        title="Order Index",
+        description="The order index of the step.",
+    )
+    annotation: WorkflowAnnotationField = None
+    label: str = Field(
+        ...,
+        title="Label",
+        description="The label of the step.",
+    )
+    inputs: List[Dict[str, Any]] = Field(
+        ...,
+        title="Inputs",
+        description="The inputs of the step.",
+    )
+
+
+class WorkflowDictEditorStep(WorkflowDictStepsExtendedBase):
+    id: int = Field(
+        ...,
+        title="ID",
+        description="The identifier of the step. It matches the index order of the step inside the workflow.",
+    )
+    name: Optional[str] = Field(
+        None,
+        title="Name",
+        description="The descriptive name of the module or step.",
+    )
+    inputs: Optional[List[Dict[str, Any]]] = Field(
+        None,
+        title="Inputs",
+        description="The inputs of the step.",
+    )
+    config_form: Optional[Dict[str, Any]] = Field(
+        None,
+        title="Config Form",
+        description="The configuration form for the step.",
+    )
+    annotation: WorkflowAnnotationField
+    uuid: Optional[UUID4] = Field(
+        None,
+        title="UUID",
+        description="Universal unique identifier of the workflow.",
+    )
+    tooltip: Optional[str] = Field(
+        None,
+        title="Tooltip",
+        description="The tooltip for the step.",
+    )
+    input_connections: Optional[Dict[str, Any]] = Field(
```

Comment:
Is there a reason you're not using `InputConnection` as you do for WorkflowDictExportStep ?